File: 006.ZigZag Conversion.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        zigzag = []
        for i in range(numRows):
            zigzag.append([])

        i = col = 0
        while i < len(s):
            if col % 2 == 1 and numRows != 2:
                inx = numRows - 2
                while inx > 0 and i < len(s):
                    zigzag[inx].append(s[i])
                    i += 1
                    inx -= 1
            else:
                for row in range(numRows):
                    if i < len(s):
                        zigzag[row].append(s[i])
                        i += 1
            col += 1
        result = ""
        for row in zigzag:
            for item in row:
                result += item
        return result

File: 006.ZigZag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        vector<vector<char> > tmp;
        bool increase = true;
        int index = 0;
        
        if(numRows == 1)
            return s;
        
        for (int i = 0; i < numRows; i++) {
            tmp.push_back(vector<char>());
        }
        
        for (int i = 0; i < s.length(); i++) {
            tmp[index].push_back(s[i]);
            if (increase) {
                if(index < numRows-1)
                    index++;
                else {
                    increase = !increase;
                    index--;
                }
            } else {
                if (index > 0)
                    index--;
                else {
                    increase = !increase;
                    index++;
                }
            }
        }
        
        string result = "";
        for (int i = 0; i < numRows; i++) {
            for (int j = 0; j < tmp[i].size(); j++) {
                result += tmp[i][j];
            }
        }
        
        return result;
    }
};
